python/cudf/cudf/core/udf/groupby_typing.py

```diff
@@ -201,13 +261,30 @@ def resolve_idxmin(self, mod):
 for ty in SUPPORTED_GROUPBY_NUMBA_TYPES:
     _register_cuda_reduction_caller("Max", ty, ty)
     _register_cuda_reduction_caller("Min", ty, ty)
-    _register_cuda_reduction_caller("Sum", ty, ty)
-    _register_cuda_reduction_caller("Mean", ty, types.float64)
-    _register_cuda_reduction_caller("Std", ty, types.float64)
-    _register_cuda_reduction_caller("Var", ty, types.float64)
     _register_cuda_idx_reduction_caller("IdxMax", ty)
     _register_cuda_idx_reduction_caller("IdxMin", ty)
 
+_register_cuda_reduction_caller("Sum", types.int32, types.int64)
```

Comment:
It feels like the typecasting rules were hard-coded twice. Once here, and once in `GroupSum`. Is there a way to reuse a single mapping / type promotion function?

Reply:
Yeah, this is a good point. There should only be one type mapping. Let me see what I can do here.

Reply:
In the latest commit, I've refactored towards an attribute generating function which checks the existing registry of cuda functions for a match in order to return a signature. I think that makes it so that this is the only place in python where the signatures are written by hand. Let me know what you think.